AFQ/_fixes.py

```diff
@@ -215,17 +216,20 @@ def gaussian_weights(bundle, n_points=100, return_mahalnobis=False,
     for i in range(n_nodes):
         # This should come back as a 3D covariance matrix with the spatial
         # variance covariance of this node across the different streamlines,
-        # reorganized as an upper diagonal matrix for expected Mahalanobis
+        # converted to a positive semi-definite matrix if necessary
         cov = np.cov(sls[:, i, :].T, ddof=0)
-        while np.any(np.linalg.eigvals(cov) < 0):
-            cov += np.eye(cov.shape[0]) * 1e-12
+        if np.any(np.linalg.eigvals(cov) < 0):
```

Comment:
Could we please upstream this fix to DIPY as well?

Reply:
Yes, lets use it for a little bit first though, to see if it needs more checks